docs/architecture/adr-001-mint-truncation-refactor.md

```diff
@@ -0,0 +1,284 @@
+<https://github.com/osmosis-labs/osmosis/blob/0b843fcae194eb9439c3dc5fe879c47173406047/x/mint/types/keys.go#L12-L21>
+
+* for persisting truncation delta resulting from the developer rewards module
+  account across epochs
+<https://github.com/osmosis-labs/osmosis/blob/0b843fcae194eb9439c3dc5fe879c47173406047/x/mint/types/keys.go#L23-L31>
+
+#### Consequences
+
+##### Positive
+
+This is helpful for accounting for truncations and distributing them eventually,
+not necessarily in the same epoch.
+
+For example, assume that for some number of epochs our expected provisions are
+`100.6` and the actual amount distributed is `100` every epoch due to truncations.
+Then, at epoch `1`, we have a delta of `0.6`. `0.6` cannot be distributed because
+it is not an integer. So we persist it until the next epoch. Then, at
+epoch `2`, we get a delta of `1.2` (`0.6` from epoch `1` and `0.6` from
+epoch `2`). Now, `1` can be distributed and `0.2` persisted until the next epoch.
+
+##### Negative
+
+Added complexity from handing two additional store indexes. It is mitigated by
+improving abstractions and documentation, though still present.
+
+### Decision 3
+
+#### Summary
+
+We will **decouple the developer vesting provisions from the inflation provisions**.
+
+The [draft implementation][15] makes the distinction between the developer
+provisions and the inflation provisions clearer by:
+
+* Distinctly [splitting the two provisions][19] in minter
+
+* Decoupling and handling each provision type separately:
+
+  * [inflation provisions](https://github.com/osmosis-labs/osmosis/blob/0b843fcae194eb9439c3dc5fe879c47173406047/x/mint/keeper/keeper.go#L167)
+
+  * [developer reward provisions](https://github.com/osmosis-labs/osmosis/blob/0b843fcae194eb9439c3dc5fe879c47173406047/x/mint/keeper/keeper.go#L283)
+
+#### Consequences
+
+##### Positive
+
+The above decoupling makes the abstractions clearer, reduces complexity, and
+fixes [#2025][14]. This change also makes it more intuitive to apply
+the [Decision 2](#decision-2) above by separating distribution logic to account
+for the 2 distinct indexes.
+
+##### Negative
+
+This is a large change to the core logic of the `x/mint` module, requiring more
+thorough testing and quality assurance.
+
+### Decision 4
+
+#### Summary
+
+We will **encapsulate the minting and distribution logic from `AfterEpochEnd` hook**
+into a separate function.
+
+In the [draft Implementation][15], the logic for distributing all epoch
+provisions in the `AfterEpochEnd` hook has been moved to the
+[distributeEpochProvisions][16] function.
+
+It handles distributing both [inflation provisions][17]
+and [developer vesting provisions][18].
+
+#### Consequences
+
+##### Positive
+
+This change allows for better encapsulation of all distribution logic and for
+the ability to more thoroughly unit test it.
+
+The enhanced encapsulation in turn leads to a more modular mint keeper where
+each method strives to focus on a single concern. 
+
```

Comment:
Just going off the format you have been using, does this imply there are no negatives to this new encapsulation? (Not disagreeing, just ensuring that was purposeful)

Reply:
Yes, I just could not think of any negatives :smile:

Feels like encapsulating this would only make it cleaner and more testable. 

If I'm missing any negatives, please let me know